interfaces/builtin/content_test.go

```diff
@@ -314,35 +314,129 @@ slots:
   remount options=(bind, ro) /snap/consumer/7/import/,
   umount /snap/consumer/7/import/,
   # Writable mimic /snap/producer/5
+  # .. permissions for traversing the prefix that is assumed to exist
+  # .. variant with mimic at /
+  / rw,
+  /tmp/.snap/ rw,
+  mount options=(rbind, rw) / -> /tmp/.snap/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /,
+  /tmp/.snap/*/ rw,
+  /*/ rw,
+  mount options=(rbind, rw) /tmp/.snap/*/ -> /*/,
+  /tmp/.snap/* rw,
+  /* rw,
+  mount options=(bind, rw) /tmp/.snap/* -> /*,
+  umount /tmp/.snap/,
+  umount /,
+  umount /*,
+  umount /*/,
+  # .. variant with mimic at /snap/
+  /snap/ rw,
+  /tmp/.snap/snap/ rw,
+  mount options=(rbind, rw) /snap/ -> /tmp/.snap/snap/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /snap/,
+  /tmp/.snap/snap/*/ rw,
+  /snap/*/ rw,
+  mount options=(rbind, rw) /tmp/.snap/snap/*/ -> /snap/*/,
+  /tmp/.snap/snap/* rw,
+  /snap/* rw,
+  mount options=(bind, rw) /tmp/.snap/snap/* -> /snap/*,
+  umount /tmp/.snap/snap/,
+  umount /snap/,
+  umount /snap/*,
+  umount /snap/*/,
+  # .. variant with mimic at /snap/producer/
+  /snap/producer/ rw,
+  /tmp/.snap/snap/producer/ rw,
+  mount options=(rbind, rw) /snap/producer/ -> /tmp/.snap/snap/producer/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /snap/producer/,
+  /tmp/.snap/snap/producer/*/ rw,
+  /snap/producer/*/ rw,
+  mount options=(rbind, rw) /tmp/.snap/snap/producer/*/ -> /snap/producer/*/,
+  /tmp/.snap/snap/producer/* rw,
+  /snap/producer/* rw,
+  mount options=(bind, rw) /tmp/.snap/snap/producer/* -> /snap/producer/*,
+  umount /tmp/.snap/snap/producer/,
+  umount /snap/producer/,
+  umount /snap/producer/*,
+  umount /snap/producer/*/,
+  # .. variant with mimic at /snap/producer/5/
+  /snap/producer/5/ rw,
+  /tmp/.snap/snap/producer/5/ rw,
   mount options=(rbind, rw) /snap/producer/5/ -> /tmp/.snap/snap/producer/5/,
   mount fstype=tmpfs options=(rw) tmpfs -> /snap/producer/5/,
-  mount options=(rbind, rw) /tmp/.snap/snap/producer/5/** -> /snap/producer/5/**,
+  /tmp/.snap/snap/producer/5/*/ rw,
+  /snap/producer/5/*/ rw,
+  mount options=(rbind, rw) /tmp/.snap/snap/producer/5/*/ -> /snap/producer/5/*/,
+  /tmp/.snap/snap/producer/5/* rw,
+  /snap/producer/5/* rw,
   mount options=(bind, rw) /tmp/.snap/snap/producer/5/* -> /snap/producer/5/*,
   umount /tmp/.snap/snap/producer/5/,
-  umount /snap/producer/5{,/**},
-  /snap/producer/5/** rw,
-  /snap/producer/5/ rw,
-  /snap/producer/ rw,
-  /tmp/.snap/snap/producer/5/** rw,
-  /tmp/.snap/snap/producer/5/ rw,
-  /tmp/.snap/snap/producer/ rw,
-  /tmp/.snap/snap/ rw,
-  /tmp/.snap/ rw,
+  umount /snap/producer/5/,
+  umount /snap/producer/5/*,
+  umount /snap/producer/5/*/,
   # Writable mimic /snap/consumer/7
+  # .. permissions for traversing the prefix that is assumed to exist
+  # .. variant with mimic at /
+  / rw,
```

Comment:
I think I've forgotten something-- I didn't think we allowed mimicking '/', and even if we did, isn't it guaranteed to exist because we pivot_root()d to it?

Reply:
That's true, I think this and the /etc/ rw rule in earlier comment suggest I missed something here. I assumed /etc/ rw is needed to make /etc/foo{,/} but .. that's not true